flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierHandler.java

```diff
@@ -95,4 +101,10 @@ protected void notifyAbortOnCancellationBarrier(long checkpointId) throws Except
 	protected void notifyAbort(long checkpointId, CheckpointException cause) throws Exception {
 		toNotifyOnCheckpoint.abortCheckpointOnBarrier(checkpointId, cause);
 	}
+
+	protected void markCheckpointStart(long checkpointCreationTimestamp) {
+		latestCheckpointStartDelayNanos = 1000_000 * Math.max(
```

Comment:
1_000_000, while you usually don't use a separator up to 9999, once you start, you use it consistently every 3 places.
Why storing it as nanos at all? Should we just track it as ms and only display it as nano (for consistency?). We are faking a precision that is not there.

Reply:
It's in the `ns` due to consistency with the existing method/field `getAlignmentDurationNanos`  and `checkpointAlignmentTime` metric.

Re 1_000_000, I don't think that it increases readability, does it matter?

Reply:
I found it weird to read, but it's clearly a nit.

Reply:
> It's in the `ns` due to consistency with the existing method/field `getAlignmentDurationNanos` and `checkpointAlignmentTime` metric.

I'd be okay with leaving as is. However, I don't think consistency is needed as the unit is carried in the name.

Reply:
Whether it is in the name or not, I think is a secondary issue. If we provide some stats, it's best if all have the same units.

https://ci.apache.org/projects/flink/flink-docs-stable/monitoring/metrics.html#checkpointing

however, annoying thing is, that in this class yes, all units are in nanos, but in other places it's millis (7 metrics in millis vs 2 in nanos?). More over, web ui is also using millis only...

Reply:
I guess after all, I would be leaning towards converging towards millis, so changing this new metric to millis as well... WDYT?

Reply:
Yes, I'd convert to ms. I don't see why any user profits from ns. 
If it was some per-record metric, I'd agree with nano (like in Kinesis?).
It's technically much too complicated to actually produce nano-precision given the expected usability gain.